client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx

```diff
@@ -116,6 +117,17 @@ export default function useCreatePaymentCompleteCallback( {
 					transactionResult.purchases && Object.keys( transactionResult.purchases ).pop();
 			}
 
+			// In the case when SaaS redirect URL is returned along with transaction, we want to redirect
+			// user to vendor application after successful purchase.
+			let saasProductRedirectUrl;
+			flatten( Object.values( transactionResult?.purchases || {} ) ).forEach(
+				( purchase: Purchase ) => {
+					if ( purchase?.saas_redirect_url ) {
+						saasProductRedirectUrl = purchase.saas_redirect_url;
```

Comment:
This function is not called for all completed purchases, only for those which do not include a redirect as part of the payment method (see comment at the top of this function). Do you want this behavior for all completed purchases? If so, you'll need to build it into the thank-you URL function or the pending page themselves.

The pending page has access to the receipt data; could it use that to query an endpoint for the purchase data this has here?